pkgs/by-name/we/weasis/package.nix

```
@@ -0,0 +1,85 @@
+{ lib
+, stdenv
+, fetchzip
+, jre
+, copyDesktopItems
+, makeDesktopItem
+}:
+
+let
+  # Build process uses this naming for output
+  platform = if stdenv.isLinux then "linux"
+             else if stdenv.isDarwin then "macosx"
+             else if stdenv.isWindows then "windows"
+             else throw "unsupported platform";
+  arch = if stdenv.isx86_64 then "x86-64"
+         else if stdenv.isAarch64 then "aarch64"
+         else if stdenv.isArmv7a then "armv7a"
+         else throw "unsupported arch";
```

This can be written more clearly using an attrset, specially because it seems you only use this value as `${platform}-${arch}`:

```suggestion
  systen = {
    x86_64-linux = "linux-x86-64";
    aarch64-linux = "linux-aarch64";
  }.${stdenv.system} or throw "unsupported system: ${stdenv.system}";
```